pkg/oauth/discovery/discovery.go

```diff
@@ -0,0 +1,51 @@
+package discovery
+
+import (
+	"github.com/RangelReale/osin"
+	"github.com/openshift/origin/pkg/authorization/authorizer/scope"
+	"github.com/openshift/origin/pkg/oauth/server/osinserver"
+)
+
+// OauthAuthorizationServerMetadata holds OAuth 2.0 Authorization Server Metadata used for discovery
+// https://tools.ietf.org/id/draft-ietf-oauth-discovery-04.html#rfc.section.2
+type OauthAuthorizationServerMetadata struct {
```

Comment:
I focused on the `REQUIRED` and `RECOMMENDED` metadata.  A lot of the `OPTIONAL` ones are not supported by OpenShift (for example, `jwks_uri` and `token_endpoint_auth_signing_alg_values_supported`).  Since @liggitt just added PKCE support, I could add `code_challenge_methods_supported` (but that may be out of scope and could certainly be added later).  `token_endpoint_auth_methods_supported`, `service_documentation`, `ui_locales_supported`, `op_policy_uri`, `op_tos_uri`, `protected_resources`, etc may all be valid now or in the future, but are probably out of scope as well.